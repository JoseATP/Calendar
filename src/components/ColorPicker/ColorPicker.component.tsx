/* eslint-disable react-hooks/exhaustive-deps */
import { FC, useState, useEffect } from "react";
import { Color, ColorContainer, Container, Label } from "./ColorPicker.styles";

export const colors = [
  {
    index: 0,
    color: "#C8E6C9",
  },
  {
    index: 1,
    color: "#F5DD29",
  },

  {
    index: 2,
    color: "#FFCC80",
  },
  {
    index: 3,
    color: "#EF9A9A",
  },
  {
    index: 4,
    color: "#CD8DE5",
  },
  {
    index: 5,
    color: "#5BA4CF",
  },
  {
    index: 6,
    color: "#6DECA9",
  },
  {
    index: 7,
    color: "#FF8ED4",
  },
  {
    index: 8,
    color: "#BCAAA4",
  },
];

type PickerColorProps = {
  value: string;
  onPickColor: (color: string) => void;
};

const PickerColor: FC<PickerColorProps> = ({ value, onPickColor }) => {
  const [selectedColor, setSelectedColor] = useState(value);

  const handleSelectedColor = (color: string) => {
    setSelectedColor(color);
  };

  useEffect(() => {
    onPickColor(selectedColor);
  }, [selectedColor]);

  return (
    <Container>
      <Label>Color</Label>
      <ColorContainer>
        {colors.map(({ index, color }) => (
          <Color
            data-testid={`color-${color}`}
            isSelected={selectedColor === color}
            key={index}
            color={color}
            onClick={(_) => handleSelectedColor(color)}
          />
        ))}
      </ColorContainer>
    </Container>
  );
};

export default PickerColor;
