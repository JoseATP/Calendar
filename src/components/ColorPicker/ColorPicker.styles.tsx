import styled, { css } from "styled-components";

export const Container = styled.div`
  display: flex;
  flex-direction: column;
  margin-bottom: 15px;
`;

export const ColorContainer = styled.div`
  display: flex;
  flex-wrap: wrap;
  justify-content: space-around;
  gap: 5px;
`;

export const Label = styled.span`
  margin-bottom: 15px;
`;

type ColorProps = {
  color: string;
  isSelected: boolean;
};

export const Color = styled.div<ColorProps>`
  width: 10%;
  height: 50px;
  background-color: ${({ color }) => color};
  cursor: pointer;
  border-radius: 5px;
  border: 1px solid rgba(49, 39, 37, 0.4);

  ${(props) =>
    props.isSelected &&
    css`
      border: 5px solid #101277;
    `}
`;
