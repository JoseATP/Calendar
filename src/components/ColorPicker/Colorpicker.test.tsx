import { render, screen, cleanup, fireEvent } from "@testing-library/react";
import ColorPicker, { colors } from "./ColorPicker.component";

afterEach(cleanup);

describe("ColorPicker", () => {
  let onPickColor = jest.fn();

  it("Should render ColorPicker component correctly", () => {
    const { asFragment } = render(
      <ColorPicker value="" onPickColor={onPickColor} />
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it("check onPickColor function call with a value", () => {
    render(<ColorPicker value={colors[1].color} onPickColor={onPickColor} />);

    expect(onPickColor).toHaveBeenCalledTimes(1);
    expect(onPickColor).toHaveBeenCalledWith(colors[1].color);
  });

  it("check onPickColor function call", () => {
    render(<ColorPicker value="" onPickColor={onPickColor} />);
    const color = screen.getByTestId(`color-${colors[0].color}`);

    expect(onPickColor).toHaveBeenCalledTimes(1);
    expect(onPickColor).toHaveBeenCalledWith("");

    fireEvent.click(color);

    expect(onPickColor).toHaveBeenCalledTimes(2);
    expect(onPickColor).toHaveBeenCalledWith(colors[0].color);
  });
});
