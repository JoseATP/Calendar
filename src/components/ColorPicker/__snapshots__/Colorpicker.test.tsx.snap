// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ColorPicker Should render ColorPicker component correctly 1`] = `
<DocumentFragment>
  <div
    class="sc-beySbM iXmWJf"
  >
    <span
      class="sc-dmyDGi ijvIZH"
    >
      Color
    </span>
    <div
      class="sc-guDLRT hsgeSm"
    >
      <div
        class="sc-hLQRIN koAene"
        color="#C8E6C9"
        data-testid="color-#C8E6C9"
      />
      <div
        class="sc-hLQRIN iqHqeY"
        color="#F5DD29"
        data-testid="color-#F5DD29"
      />
      <div
        class="sc-hLQRIN ewxdLE"
        color="#FFCC80"
        data-testid="color-#FFCC80"
      />
      <div
        class="sc-hLQRIN cnElvf"
        color="#EF9A9A"
        data-testid="color-#EF9A9A"
      />
      <div
        class="sc-hLQRIN fqgqSH"
        color="#CD8DE5"
        data-testid="color-#CD8DE5"
      />
      <div
        class="sc-hLQRIN hmwvBX"
        color="#5BA4CF"
        data-testid="color-#5BA4CF"
      />
      <div
        class="sc-hLQRIN dKhOhk"
        color="#6DECA9"
        data-testid="color-#6DECA9"
      />
      <div
        class="sc-hLQRIN YZWlB"
        color="#FF8ED4"
        data-testid="color-#FF8ED4"
      />
      <div
        class="sc-hLQRIN epqrTA"
        color="#BCAAA4"
        data-testid="color-#BCAAA4"
      />
    </div>
  </div>
</DocumentFragment>
`;
